nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITDeleteAzureDataLakeStorage.java

```diff
@@ -35,15 +35,82 @@
 
 public class ITDeleteAzureDataLakeStorage extends AbstractAzureDataLakeStorageIT {
 
+    private static final boolean FILE = true;
+    private static final boolean DIRECTORY = false;
+
     @Override
     protected Class<? extends Processor> getProcessorClass() {
         return DeleteAzureDataLakeStorage.class;
     }
 
+    @Test
+    public void testDeleteDirectoryWithFiles() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String filename = "testFile.txt";
+        String fileContent = "AzureFileContent";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        createDirectoryAndUploadFile(directory, filename, fileContent);
+
+        // WHEN
+        // THEN
+        testSuccessfulDelete(fileSystemName, directory, filename, inputFlowFileContent, inputFlowFileContent, DIRECTORY);
+    }
+
+    @Test
+    public void testDeleteEmptyDirectoryAsFolder() {
+        // GIVEN
+        String directory = "TestDirectory";
+        String inputFlowFileContent = "InputFlowFileContent";
+
+        createDirectory(directory);
+
+        // WHEN
+        // THEN
+        testSuccessfulDelete(fileSystemName, directory, null, inputFlowFileContent, inputFlowFileContent, DIRECTORY);
+    }
+
+    @Test
+    public void testDeleteSubdirectory() {
+        // GIVEN
+        String parentDirectory = "TestParentDirectory";
+        String childDirectory = "TestParentDirectory/TestChildDirectory";
+        String filename = "testFile.txt";
+        String fileContent = "AzureFileContent";
```

Comment:
`fileContent` is not used. It could be deleted or as a better option, a file could be uploaded in order to test deleting a non-empty subdirectory.
Also for the following test case.